File: HolyDay/Models/Prayer/PrayerStep.swift
//
//  PrayerStep.swift
//  HolyDay
//
//  Created by Matthias Cadet on 13/05/2026.
//

import Foundation
import SwiftUI

struct PrayerStep: Identifiable {
  let id: UUID
  let title: String
  let description: String
  let icon: String
  let colorName: String
  let order: Int

  var color: Color {
    AppTheme.color(for: colorName)
  }

  init(
    id: UUID = UUID(), title: String, description: String, icon: String, colorName: String,
    order: Int
  ) {
    self.id = id
    self.title = title
    self.description = description
    self.icon = icon
    self.colorName = colorName
    self.order = order
  }

  static let defaultSteps: [PrayerStep] = [
    PrayerStep(
      title: String(localized: "step.adoration.title"),
      description: String(localized: "step.adoration.description"),
      icon: "hands.sparkles",
      colorName: "adorationPurple",
      order: 1
    ),
    PrayerStep(
      title: String(localized: "step.confession.title"),
      description: String(localized: "step.confession.description"),
      icon: "heart.circle",
      colorName: "confessionBlue",
      order: 2
    ),
    PrayerStep(
      title: String(localized: "step.thanksgiving.title"),
      description: String(localized: "step.thanksgiving.description"),
      icon: "star.circle",
      colorName: "thanksgivingGold",
      order: 3
    ),
    PrayerStep(
      title: String(localized: "step.supplication.title"),
      description: String(localized: "step.supplication.description"),
      icon: "bubble.left.and.bubble.right",
      colorName: "supplicationGreen",
      order: 4
    ),
  ]
}

File: android/app/src/main/java/com/matthiascadet/holyday/data/model/PrayerStep.kt
package com.matthiascadet.holyday.data.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Chat
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Star
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.matthiascadet.holyday.R
import com.matthiascadet.holyday.ui.theme.AppTheme
import java.util.UUID

/** Une des 4 étapes de la prière guidée ACTS (Adoration, Confession, Actions de grâce, Supplication). */
data class PrayerStep(
    val id: UUID = UUID.randomUUID(),
    val titleRes: Int,
    val descriptionRes: Int,
    val icon: ImageVector,
    val colorName: String,
    val order: Int,
) {
    @Composable
    fun color(): Color = AppTheme.colorFor(colorName)

    companion object {
        val defaultSteps: List<PrayerStep> = listOf(
            PrayerStep(
                titleRes = R.string.step_adoration_title,
                descriptionRes = R.string.step_adoration_description,
                icon = Icons.Filled.AutoAwesome,
                colorName = "adorationPurple",
                order = 1,
            ),
            PrayerStep(
                titleRes = R.string.step_confession_title,
                descriptionRes = R.string.step_confession_description,
                icon = Icons.Filled.Favorite,
                colorName = "confessionBlue",
                order = 2,
            ),
            PrayerStep(
                titleRes = R.string.step_thanksgiving_title,
                descriptionRes = R.string.step_thanksgiving_description,
                icon = Icons.Filled.Star,
                colorName = "thanksgivingGold",
                order = 3,
            ),
            PrayerStep(
                titleRes = R.string.step_supplication_title,
                descriptionRes = R.string.step_supplication_description,
                icon = Icons.AutoMirrored.Filled.Chat,
                colorName = "supplicationGreen",
                order = 4,
            ),
        )
    }
}
